src/nupic/types/Types.h

```diff
@@ -218,6 +225,24 @@ typedef  long long           NTA_Int64;
  */
 typedef  unsigned long long  NTA_UInt64;
 
+#elif defined(NTA_ARCH_64) && defined(NTA_OS_WINDOWS)
+/**
+ * Represents a 32-bit signed integer.
+ */
+typedef  long                NTA_Int32;
+/**
+ * Represents a 32-bit unsigned integer.
+ */
+typedef  unsigned long       NTA_UInt32;
+/**
+ * Represents a 64-bit signed integer.
+ */
+typedef  long long           NTA_Int64;
+/**
+ * Represents a 64-bit unsigned integer.
+ */
+typedef  unsigned __int64    NTA_UInt64;
```

Comment:
@scottpurdy 

Should I also replace: 

`typedef unsigned __int64 NTA_UInt64;`

to:

`typedef unsigned long long NTA_UInt64;` ?